### DISCUSSION TITLE
Accessing records' data using variable

### DISCUSSION THREAD

**Discussion by Tom-Boscher** [nushell/nushell#10472] [Q&A] [answered]
I have a record :
```
let my_record = {
    aaa: bbb
    ccc: ddd
    eee: fff
}
```
I want to access to "bbb" using a variable `let my_variable = "aaa"`
I tried:
```
let this_should_be_aaa = $my_record.$my_variable
```
but I get an error `cannot find column '$my_variable'`

**Comment by ghost** [comment, marked as answer]
you can't use variables in `.` - you can use `$my record | get $my_variable`

**Reply by Tom-Boscher** [reply]
Thank you, as it works like a table indeed.

**Reply by eugenesvk** [reply]
> you can't use variables in `.`

is there some deep reason behind this most convenient accessor being blocked or could this be a future reason?

**Reply by ghost** [reply]
there is some related discussion in here <https://github.com/nushell/nushell/issues/8480>.  
i think it might be due to questions like is `let a = "b.c"; $foo.$a` = `$foo.b.c` or `$foo."b.c"`, etc.
